frontend/src/main/scala/akkaviz/frontend/components/HierarchyPanel.scala

```diff
@@ -0,0 +1,90 @@
+package akkaviz.frontend.components
+
+import akkaviz.frontend.FrontendUtil
+import org.querki.jquery.JQueryStatic
+import org.scalajs.dom._
+
+import scala.scalajs.js.Dictionary
+import scalatags.JsDom.all._
+
+class HierarchyPanel(detailsOpener: (String) => Unit) extends Component {
+
+  private[this] val $ = JQueryStatic
+
+  private[this] val seenActors: Dictionary[Unit] = Dictionary()
+
+  private[this] val ActorAttr = "actor-path".attr
+
+  private[this] val hierarchy = div(
+    cls := "actor-tree",
+    ActorAttr := "root",
+    ul()
+  ).render
+
+  private[this] def node(actorRef: String) = li(
+    ActorAttr := actorRef,
+    span(
+      i(cls := "glyphicon glyphicon-leaf"),
+      shortName(actorRef),
+      title := actorRef,
+      "data-target".attr := s"""[actor-path="$actorRef"]>ul""",
+      "data-toggle".attr := "collapse",
+      onclick := { () => nodeClicked(actorRef) }
+    ),
+    a(
+      "(details)",
+      href := "#",
+      onclick := { () => detailsOpener(actorRef) }
+    ),
+    ul(
+      cls := "collapse"
+    )
+  )
+
+  private[this] def nodeClicked(ref: String) = {
+    val node = $(s"""[actor-path="$ref"]""")
+    val nodeUl = node.find("ul").first
+    val isEmpty = nodeUl.find("ul").first.length == 0
+    if (!isEmpty) {
+      val isExpanded = node.find("ul").first.hasClass("in")
+      if (isExpanded)
+        node.find("span>i").first.removeClass("glyphicon-minus").addClass("glyphicon-plus")
+      else
+        node.find("span>i").first.removeClass("glyphicon-plus").addClass("glyphicon-minus")
+    }
+  }
+
+  private[this] def shortName(ref: String) = {
+    ref.stripPrefix("akka://").split("/").last
+  }
+
+  def insert(ref: String): Unit = innerInsert(ref.stripSuffix("/"))
+
+  def insert(refs: Iterable[String]): Unit = refs.foreach(insert)
+
+  private[this] def innerInsert(ref: String): Unit = {
+    if (!exists(ref)) {
+      val parentRef = FrontendUtil.parent(ref)
+      parentRef.foreach(innerInsert)
+      insertSorted(parentRef.getOrElse("root"), ref)
+      seenActors.update(ref, ())
+    }
+  }
+
+  private[this] def insertSorted(parentRef: String, ref: String): Unit = {
+    val parentUl = $(s"""[actor-path="$parentRef"]>ul""").toArray.head
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://www.codacy.com/app/lustefaniak/akka-viz/file/2204579680/issues/source?bid=3121282&fileBranchId=3170129#l75)